src/optimagic/optimizers/iminuit_migrad.py

```diff
@@ -0,0 +1,154 @@
+        n_fun_evals=minuit_result.nfcn,
+        n_jac_evals=minuit_result.ngrad,
+        n_hess_evals=None,
+        n_iterations=minuit_result.nfcn,
+        status=None,
+        jac=None,
+        hess=None,
+        hess_inv=np.array(minuit_result.covariance),
+        max_constraint_violation=None,
+        info=None,
+        history=None,
+    )
+    return res
+
+
+def _convert_bounds_to_minuit_limits(
+    lower_bounds: Optional[NDArray[np.float64]],
+    upper_bounds: Optional[NDArray[np.float64]],
+) -> list[tuple[Optional[float], Optional[float]]]:
+    """Convert optimization bounds to Minuit-compatible limit format.
+
+    Transforms numpy arrays of bounds into List of tuples as expected by iminuit.
+    Handles special values like np.inf, -np.inf, and np.nan by converting
+    them to None where appropriate, as required by Minuit's limits API.
+
+    Parameters
+    ----------
+    lower_bounds : Optional[NDArray[np.float64]]
+        Array of lower bounds for parameters.
+    upper_bounds : Optional[NDArray[np.float64]]
+        Array of upper bounds for parameters.
+
+    Returns
+    -------
+    list[tuple[Optional[float], Optional[float]]]
+        List of (lower, upper) limit tuples in Minuit format, where:
+        - None indicates unbounded (equivalent to infinity)
+        - Float values represent actual bounds
+
+    Notes
+    -----
+    Minuit expects bounds as tuples of (lower, upper) where:
+    - `None` indicates no bound (equivalent to -inf or +inf)
+    - A finite float value indicates a specific bound
+    - Bounds can be asymmetric (e.g., one side bounded, one side not)
+
+    """
+    if lower_bounds is None or upper_bounds is None:
+        return []
+
+    return [
+        (
+            None if np.isneginf(lower) or np.isnan(lower) else float(lower),
+            None if np.isposinf(upper) or np.isnan(upper) else float(upper),
+        )
+        for lower, upper in zip(lower_bounds, upper_bounds, strict=True)
+    ]
+
+
+def _set_minuit_limits(
+    m: Minuit, bounds: list[tuple[Optional[float], Optional[float]]]
+) -> None:
+    """Set parameter limits on a Minuit minimizer instance.
+
+    Applies the converted bounds to an iminuit.Minuit object. Minuit expects
+    parameter limits as tuples of (lower, upper) for each parameter, where
+    None indicates an unbounded direction.
+
+    Parameters
+    ----------
+    m : Minuit
+        The iminuit minimizer instance to configure.
+    bounds : list[tuple[Optional[float], Optional[float]]]
+        List of parameter bounds as (lower, upper) tuples in Minuit format.
+        For each tuple:
+        - (None, None): Fully unbounded parameter
+        - (value, None): Lower bound only
+        - (None, value): Upper bound only
+        - (min, max): Two-sided constraint
+
+    """
+    for i, (lower, upper) in enumerate(bounds):
+        if lower is not None or upper is not None:
+            m.limits[i] = (lower, upper)
```

Comment:
I would do this directly in `_solve_internal_problem` to avoid having an impure function here. By impure I mean that the function changes the `Minuit` object in place and returns nothing.